packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesFileDownloader.m

```diff
@@ -169,7 +180,7 @@ - (void)_downloadDataWithRequest:(NSURLRequest *)request
   NSURLSessionDataTask *task = [_session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
     if (!error && [response isKindOfClass:[NSHTTPURLResponse class]]) {
       NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
-      if (httpResponse.statusCode != 200) {
+      if (httpResponse.statusCode < 200 || httpResponse.statusCode >= 300) {
```

Comment:
If the server responds with a 304 (etag match), does NSURLSession automatically rewrite it to look like a 200?

Reply:
yes, it uses the entire cached response including the status code.

Reply:
Nice, that makes this simple to deal with.